doc/tutorials/klondike/app/lib/step4/components/card.dart

```diff
@@ -0,0 +1,282 @@
+        _drawSprite(canvas, suitSprite, 0.7, 0.4);
+        _drawSprite(canvas, suitSprite, 0.3, 0.2, rotate: true);
+        _drawSprite(canvas, suitSprite, 0.7, 0.2, rotate: true);
+        _drawSprite(canvas, suitSprite, 0.5, 0.3, rotate: true);
+        _drawSprite(canvas, suitSprite, 0.3, 0.4, rotate: true);
+        _drawSprite(canvas, suitSprite, 0.7, 0.4, rotate: true);
+        break;
+      case 11:
+        _drawSprite(canvas, suit.isRed ? redJack : blackJack, 0.5, 0.5);
+        break;
+      case 12:
+        _drawSprite(canvas, suit.isRed ? redQueen : blackQueen, 0.5, 0.5);
+        break;
+      case 13:
+        _drawSprite(canvas, suit.isRed ? redKing : blackKing, 0.5, 0.5);
+        break;
+    }
+  }
+
+  void _drawSprite(
+    Canvas canvas,
+    Sprite sprite,
+    double relativeX,
+    double relativeY, {
+    double scale = 1,
+    bool rotate = false,
+  }) {
+    if (rotate) {
+      canvas.save();
+      canvas.translate(size.x / 2, size.y / 2);
+      canvas.rotate(pi);
+      canvas.translate(-size.x / 2, -size.y / 2);
+    }
+    sprite.render(
+      canvas,
+      position: Vector2(relativeX * size.x, relativeY * size.y),
+      anchor: Anchor.center,
+      size: sprite.srcSize.scaled(scale),
+    );
+    if (rotate) {
+      canvas.restore();
+    }
+  }
+
+  //#endregion
+
+  //#region Dragging
+
+  @override
+  void onDragStart(DragStartEvent event) {
+    if (pile?.canMoveCard(this) ?? false) {
+      _isDragging = true;
+      priority = 100;
+      if (pile is TableauPile) {
+        attachedCards.clear();
+        final extraCards = (pile! as TableauPile).cardsOnTop(this);
+        for (final card in extraCards) {
+          card.priority = attachedCards.length + 101;
+          attachedCards.add(card);
+        }
+      }
+    }
+  }
+
+  @override
+  void onDragUpdate(DragUpdateEvent event) {
+    if (!_isDragging) {
+      return;
+    }
+    final cameraZoom = (findGame()! as FlameGame)
+        .firstChild<CameraComponent>()!
+        .viewfinder
+        .zoom;
+    final delta = event.delta / cameraZoom;
+    position.add(delta);
+    attachedCards.forEach((card) => card.position.add(delta));
+  }
+
+  @override
+  void onDragEnd(DragEndEvent event) {
+    if (!_isDragging) {
+      return;
+    }
+    _isDragging = false;
+    final dropPiles = parent!
+        .componentsAtPoint(position + size / 2)
+        .whereType<Pile>()
+        .toList();
+    if (dropPiles.isNotEmpty) {
+      if (dropPiles.first.canAcceptCard(this)) {
+        pile!.removeCard(this);
+        dropPiles.first.acquireCard(this);
+        if (attachedCards.isNotEmpty) {
+          attachedCards.forEach((card) => dropPiles.first.acquireCard(card));
+          attachedCards.clear();
+        }
+        return;
+      }
+    }
+    pile!.returnCard(this);
+    if (attachedCards.isNotEmpty) {
+      attachedCards.forEach((card) => pile!.returnCard(card));
+      attachedCards.clear();
+    }
+  }
+
+  //#endregion
```

Comment:
This file ended up being quite big, could we extract some parts of it to different files?

Reply:
Agree, the class is getting kinda long -- that's why I've added some `#region`s in it. The rendering part is explained in the previous chapter of the tutorial, in this chapter we only deal with the dragging behavior. So, hopefully, it's not too overwhelming for the user.

Reply:
That would be simpler for us, but it wouldn't be simpler for a beginner of Flame to read I think.

Reply:
Yeah, the region helps on IDEs that support it, but users reading on github would not benefit from that.

Reply:
Agree that the region's probably doesn't help that much for most beginner's reading this, but I also don't think that splitting it up would help. I think we should just keep it like it is. :)

Reply:
Alright